Assets/00_Content/Scripts/Interactables/ItemSlot.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Interactables {
+	public class ItemSlot : MonoBehaviour {
+
+		[SerializeField] private LayerMask pickUpLayer;
+
+		private bool hasItemInSlot = false;
+
+		public bool HasItemInSlot => hasItemInSlot;
+
+		private void FixedUpdate() {
```

Comment:
It feels a bit too much checking this every FixedUpdate, could it be reduced using enter/exit or by "giving" it a pickup and let it handle itself via events or something?

Reply:
I wish, but as we are disabling and enabling the colliders on picked up objects the onEnter/Exit will not behave as expected.